### DISCUSSION TITLE
Why doesn't Uno Wasm work anymore on github pages?

### DISCUSSION THREAD

**Discussion by Ibrahim12404** [unoplatform/uno#16527] [General]
I just got into trying Uno's Wasm thingy. I am a hobbyist trying to build for myself a simple blog engine using Uno's wasm. I chose github pages to deploy it to. Cool story.

I found a github actions yaml to compile uno wasm on the fly and deploy it to github pages in github. So, I did exactly as I am supposed to. I opened an uno wasm project in vscode and uploaded it to a github repo. Then, I ran the yaml. It compiles and deploys just fine.

The problem is, uno wasm from 3 years ago with .net 2.0 and old uno version works just fine with github pages, but, the newer .net 8.0 and latest uno simply doesn't work. How come? What am I missing?

This is the one that works https://github.com/Ibrahim12404/UnoPagesFork (It is a fork from this guy here https://dev.to/kurtmkurtm/uno-platform-net-webassembly-apps-on-github-pages-5873. I compiled it and deployed it just now and it works.)

Test it here: https://ibrahim12404.github.io/UnoPagesFork/

This is the one that doesn't work https://github.com/Ibrahim12404/UnoTest (I copied the yaml from the working example that uses older dotnet and older uno, and, then I updated the yaml to latest dotnet and uno. Build and deploy work 100% correctly. But, the website doesn't work. There are too many 404 errors).

See here: https://ibrahim12404.github.io/UnoTest/

It seems that uno wasm back in the days worked just fine on github pages, but, now it doesn't anymore? Because it is the exact same yaml on github actions, both with just new uno wasm app from the templates, only the one working is old version and the one not working is new one.

Anyway, what are the steps to put my uno wasm app on github pages? There is some instructions on putting it on Azure. But, I am not interested in giving my credit card info anywhere. I can't find anything on the web or the search on here on how to put uno wasm app on github pages.

Thanks for the help

**Comment by nor0x** [comment]
seems that there is something wrong with the base path of your web app - as some resources could not be found. These resources (i.e. require.js) are located in the package directory - in your case the reference from the HTML code is at the root `https://ibrahim12404.github.io` and not `https://ibrahim12404.github.io/UnoTest/`
![image](https://github.com/unoplatform/uno/assets/3210391/55faceb1-a919-4fd2-ac29-387219fe8812)

at the UnoPagesFork the package link is referenced correctly:
![image](https://github.com/unoplatform/uno/assets/3210391/c0eb064e-df87-48ba-a555-b999d01d4358)

**Reply by Ibrahim12404** [reply]
Yeah, I see that. But, if I add a . before the / in the srcs in index.html, I then get 404 errors further down in some of the javascripts again! I don't know how to fix this. How to let github know that it should see the base path correctly? Because in previous uno versions, it added the "." before "/" itself (i.e., it referenced the packages correctly, which it doesn't anymore). It works on local IIS servers with the way it is tho, but, not on github pages.
<img width="864" alt="123" src="https://github.com/unoplatform/uno/assets/97126964/5751afeb-d937-467c-abff-49bec0183441">
<img width="903" alt="125" src="https://github.com/unoplatform/uno/assets/97126964/75bd4b10-92da-4adf-aff9-80e127fcee82">

Somebody help me please. I am new to this thing. Does github pages not support uno wasm apps or something, or am I being stupid?

**Reply by jeromelaban** [reply]
When hosting in github pages, you need to specify the base path: https://platform.uno/docs/articles/external/uno.wasm.bootstrap/doc/features-deep-linking.html

**Reply by Ibrahim12404** [reply]
Nice! I just added -p:WasmShellWebAppBasePath=./ in the dotnet build command and it works like a charm!
